activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -662,6 +662,8 @@ def translate_exception(exception, message:, sql:, binds:)
             InvalidForeignKey.new(message, sql: sql, binds: binds, connection_pool: @pool)
           elsif exception.message.match?(/called on a closed database/i)
             ConnectionNotEstablished.new(exception, connection_pool: @pool)
+          elsif exception.is_a?(::SQLite3::BusyException)
+            Deadlocked.new(message, sql: sql, binds: binds, connection_pool: @pool)
```

Comment:
`Deadlocked` has a meaning inside transaction, re-using this class could have bad repercussions.

I think `StatementTimeout` could work.

Reply:
@byroot Thank you! Updated.